### PR TITLE
Replace rcf/kll model with ThresholdedRandomCutForest

### PR DESCRIPTION
**Pull request by kaituo**
Note: since there are inter-file references, I split a big PR into multiple smaller PRs based on files to save time (1.1 cut off is due this Friday). The build will fail due to missing dependencies. I will use that PR just for review. will not merge it. Will have a big PR in the end and merge once after all review PRs get approved. For the complete code, check https://github.com/kaituo/anomaly-detection-1/tree/thresholdRCF2

### Description
This PR does the following things:
1. added the new binaries of RCF library.
2. removed various places of kll threshold model.  Only keep it in ModelManager and CheckpointDao for backward compatibility purposes.  Kll threshold model is still needed because 1)old nodes can send threshold result messages to new nodes; 2)we need to know how to convert from kll threshold model and rcf model to ThresholdedRandomCutForest.
3. Removed the constructor of EntityModel that takes rcf and threshold models as we won't run old rcf/kll models in HCAD and EntityModel is only used in HCAD.  
4. Remove model partitioning logic in single-stream workflow.  We have already disabled the code path before.  Removing the code for easier maintenance.
5. Change to use ThresholdedRandomCutForest in preview API.

Testing done:
1. Manual bwc testing
2. real-time single stream and HCAD detectors testing
3. Make sure all unit/integ/system tests pass.
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/main/java/org/opensearch/ad/ml/ModelManager.java [hunk 1]
```diff
@@ -707,51 +558,16 @@ public void maintenance(ActionListener<Void> listener) {
             .compact(true)
             .precision(Precision.FLOAT_32)
             .boundingBoxCacheFraction(AnomalyDetectorSettings.BATCH_BOUNDING_BOX_CACHE_RATIO)
-            // same with dimension for opportunistic memory saving
-            .shingleSize(shingleSize)
+            // for external shingling, rcf does not recognize shingle
+            .shingleSize(1)
```
Comment (wnbts) [inline src/main/java/org/opensearch/ad/ml/ModelManager.java:562]: does this affect singles and multis?

Comment (kaituo) [reply src/main/java/org/opensearch/ad/ml/ModelManager.java:562]: This affects single-stream real time detectors.  HCAD models' configuration is in EntityColdStarter.

Comment (wnbts) [reply src/main/java/org/opensearch/ad/ml/ModelManager.java:562]: is this getPreviewResults called by multis?

Comment (kaituo) [reply src/main/java/org/opensearch/ad/ml/ModelManager.java:562]: Sorry, I was looking at the wrong places.  Yes, this affects both single stream/HCAD preview.

File: src/main/java/org/opensearch/ad/ml/EntityModel.java [hunk 2]
```diff
@@ -32,24 +32,14 @@
...
 
-    public EntityModel(Entity entity, Queue<double[]> samples, RandomCutForest rcf, ThresholdingModel threshold) {
-        this.entity = entity;
-        this.samples = samples;
-        this.rcf = rcf;
-        this.threshold = threshold;
-    }
-
     /**
      * Constructor with ERCF.
```
Comment (sudiptoguha) [inline src/main/java/org/opensearch/ad/ml/EntityModel.java:44]: Change to tRCF? Not urgent/not important right now.

Comment (kaituo) [reply src/main/java/org/opensearch/ad/ml/EntityModel.java:44]: changed.

File: src/main/java/org/opensearch/ad/ml/ModelManager.java (hunk 1 above)
Comment (sudiptoguha) [reply src/main/java/org/opensearch/ad/ml/ModelManager.java:562]: OK. So this shingleSize = 1 for the tRCF is a signal that shingling is external. So we are using external shingling in the HCAD preview as well? We are using internal shingling in HCAD right?

Comment (sudiptoguha) [reply src/main/java/org/opensearch/ad/ml/ModelManager.java:562]: (information question so that we are all on same page)

Comment (kaituo) [reply src/main/java/org/opensearch/ad/ml/ModelManager.java:562]: yes, we are using external shingling in HCAD preview, but internal shingling in HCAD real time detectors.

**Review by sudiptoguha — APPROVED** [review]

**Review by wnbts — APPROVED** [review]

File: src/main/java/org/opensearch/ad/ml/ModelManager.java (hunk 1 above)
Comment (sudiptoguha) [reply src/main/java/org/opensearch/ad/ml/ModelManager.java:562]: Wait this shingleSize will be passed to the forest, so we would have shingleSize as 1 again.

**Pull request closed without merging**